core/tools/blueos_startup_update/blueos_startup_update.py

```diff
@@ -545,6 +545,57 @@ def configure_network_manager() -> bool:
     return True
 
 
+def check_available_space(required_mb: int) -> bool:
+    """Check if there is enough space in disk for the required megabytes"""
+    try:
+        stats = os.statvfs("/")
+        available_mb = stats.f_bavail * stats.f_frsize / (1024 * 1024)
+        return available_mb >= required_mb
+    except Exception as error:
+        logger.error(f"Failed to check available space: {error}")
+        return False
+
+
+def update_swap_size() -> bool:
+    """Updates the swap size if there is enough space available"""
+    logger.info("Checking and updating swap size...")
+
+    config_file = "/etc/dphys-swapfile"
+
+    if check_available_space(1024):
+        desired_size = 1024
+    elif check_available_space(512):
+        desired_size = 512
```

Comment:
maybe we should require 30% more space than what we aim for the swap file

Reply:
good idea